editor/src/clj/editor/model_util.clj

```diff
@@ -0,0 +1,141 @@
+                              :ex-data (ex-data exception)))
+                  nil)))
+            (graphics.types/default-attribute-value-array semantic-type element-type))]
+
+    (case attribute-transform
+      (:attribute-transform-none)
+      (attribute/make-attribute-value-binding value-array element-type location)
+
+      (:attribute-transform-normal :attribute-transform-world)
+      (let [transform-render-arg-key (graphics.types/attribute-transform-render-arg-key attribute-transform)
+            w-component (graphics.types/attribute-transform-w-component attribute-transform)]
+        (attribute/make-transformed-attribute-value-binding value-array element-type transform-render-arg-key w-component location)))))
+
+(defn- make-attribute-render-arg-binding
+  [render-arg-key attribute-info]
+  (let [{:keys [data-type location normalize vector-type]} attribute-info
+        element-type (graphics.types/make-element-type vector-type data-type normalize)]
+    (attribute/make-attribute-render-arg-binding render-arg-key element-type location)))
+
+(defn- make-render-arg-attribute-binding-entries
+  [attribute-infos render-arg-key]
+  (->> attribute-infos
+       (e/map
+         (fn [{:keys [name-key] :as attribute-info}]
+           {:pre [(keyword? name-key)]}
+           (pair name-key
+                 (make-attribute-render-arg-binding render-arg-key attribute-info))))))
+
+(defn- make-attribute-binding-entries
+  [attribute-infos attribute-buffers name-key->attribute-bytes scene-node-id]
+  (let [attribute-buffer-count (count attribute-buffers)]
+    (->> attribute-infos
+         (e/map-indexed
+           (fn [^long channel {:keys [name-key] :as attribute-info}]
+             {:pre [(keyword? name-key)]}
+             ;; Use attribute buffers from the mesh when available. Otherwise,
+             ;; use the overridden attribute value from the referencing
+             ;; component. If not overridden by the component, use the value
+             ;; specified for the attribute in the Material. Finally, if none of
+             ;; the above are available, use a suitable default value inferred
+             ;; from the semantic-type.
+             (pair name-key
+                   (if (< channel attribute-buffer-count)
+                     (let [attribute-buffer (attribute-buffers channel)]
+                       (make-attribute-buffer-binding attribute-buffer attribute-info scene-node-id))
+                     (let [attribute-bytes (or (name-key->attribute-bytes name-key) ; From the referencing component.
+                                               (:bytes attribute-info))] ; From the Material.
+                       (make-attribute-value-binding attribute-bytes attribute-info)))))))))
+
+(defn make-attribute-bindings
+  ([semantic-type->attribute-buffers combined-attribute-infos]
+   (make-attribute-bindings semantic-type->attribute-buffers combined-attribute-infos {} nil))
```

Comment:
This place passes `nil` as a `scene-node-id`, but as it gets passed along, at some point the code asserts `(gt/node-id? scene-node-id)` (in `make-transformed-attribute-buffer`)